plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3FileSystem.java

```diff
@@ -1597,6 +1601,7 @@ public void write(int b)
         public void write(byte[] bytes, int offset, int length)
                 throws IOException
         {
+            ensureExtraBytesCapacity(length);
             while (length > 0) {
```

Comment:
you can do:
```
ensureExtraBytesCapacity(min(partSize - bufferSize, length));
```
in a loop.

Reply:
yes sure, i thought it is less optimal to do it in the loop but ok

Reply:
It shouldn't matter as `partSize` is min 5MB